core-bundle/src/Routing/RouteProvider.php

```diff
@@ -488,9 +491,15 @@ private function findPages(array $candidates): array
             $conditions[] = 'tl_page.alias IN ('.implode(',', $aliases).')';
         }
 
+        $options = [];
+
+        if (!empty($set)) {
+            $options['order'] = Database::getInstance()->findInSet('alias', $set);
```

Comment:
I don't really like that we have to work with the `Database` object here. Maybe we can use https://github.com/beberlei/DoctrineExtensions here?